pinot-filesystem/src/main/java/com/linkedin/pinot/filesystem/LocalPinotFS.java

```diff
@@ -97,7 +97,8 @@ public boolean move(URI srcUri, URI dstUri, boolean overwrite) throws IOExceptio
   public boolean copy(URI srcUri, URI dstUri) throws IOException {
     File srcFile = new File(decodeURI(srcUri.getRawPath()));
     File dstFile = new File(decodeURI(dstUri.getRawPath()));
-    if (dstFile.exists()) {
+    // delete dst only if dst isn't under src.
+    if (!dstFile.getCanonicalPath().startsWith(srcFile.getCanonicalPath()) && dstFile.exists()) {
```

Comment:
Is it valid to delete dst if its a directory?

If someone attempts to copy name1 to dst/name1, where the source name1 is a directory and dst/name1 is a file, shouldn't the copy fail?

Reply:
In your case, src is name1 and dst is dst/name1.
So the condition `!dstFile.getCanonicalPath().startsWith(srcFile.getCanonicalPath())` is satisfied and so is`dstFile.exists()`. Thus, the previous dst will be deleted.

Reply:
I don't think previous dst/name1 should be deleted. Will be good to review this and see if we should honor this contract:
If name1 is a folder and dst/name1 is a file, operation should fail.
If name1 is a folder and dst/name1 is a folder, src folder will get copied under dst/name1.
If name1 is a file and dst/name1 is a file, dst/name1 will get overwritten.
If name1 is a file and dst/name1 is a folder, src file will get copied under dst/name1.

Reply:
That's a good question. @jenniferdai do you have any comments on it?
I've modified the code to keep the original destination dir if it exists.

Reply:
You just need to make sure that all code references to these methods have the same contract as well.

Reply:
Discussed with @jenniferdai offline. She's happy with the changes @sunithabeeram proposed.

Reply:
@jenniferdai I've verified that all code references to these methods have the same contract.